packages/jspsych/tests/extensions/extensions.test.ts

```diff
@@ -4,188 +4,242 @@ import { JsPsych, initJsPsych } from "../../src";
 import { pressKey } from "../utils";
 import testExtension from "./test-extension";
 
-let jsPsych: JsPsych;
-
 jest.useFakeTimers();
 
 // https://github.com/jspsych/jsPsych/projects/6#card-64825201
-describe.skip("jsPsych.extensions", () => {
-  beforeEach(() => {
-    jsPsych.extensions.test = testExtension;
-  });
+describe("jsPsych.extensions", () => {
 
-  test("initialize is called at start of experiment", () => {
-    var initFunc = jest.spyOn(jsPsych.extensions.test, "initialize");
+  test("initialize is called at start of experiment", async () => {
 
-    var timeline = [{ type: htmlKeyboardResponse, stimulus: "foo" }];
-
-    jsPsych = initJsPsych({
-      timeline,
-      extensions: [{ type: "test" }],
+    let jsPsych = initJsPsych({
+      on_finish: () => {
+        expect(initFunc).toHaveBeenCalled();
+      },
```

Comment:
How does this approach differ from awaiting `jsPsych.run()` and then asserting initFunc to have been called?